llvm/lib/Transforms/Coroutines/CoroSplit.cpp

```diff
@@ -2108,12 +2108,6 @@ PreservedAnalyses CoroSplitPass::run(LazyCallGraph::SCC &C,
   if (Coroutines.empty() && PrepareFns.empty())
     return PreservedAnalyses::all();
 
-  if (Coroutines.empty()) {
```

Comment:
Why is this change NFC?

Reply:
Because the next lines reads "for each coroutine in Coroutines". It's the same condition the if statement is checking for, then execute the exact same logic.

Reply:
But it reads the PrepareFns, are they the same thing?

Reply:
Oh sorry. I meant L2118 here. `for (LazyCallGraph::Node *N : Coroutines) {`

With and without this block, the executed logic is identical. With this block, this PrepareFns gets replaced with `replaceAllPrepares` on L2113. Without this block, if the Coroutines collection happens to be empty, it goes to the L2118 loop without any iteration, it goes to L2146 to perform the same logic.